### DISCUSSION TITLE
Why does an error occur when using prisma db pull?

### DISCUSSION THREAD

**Discussion by sihun0105** [prisma/prisma#21478] [Q&A] [answered]
### Question

OS : CentOs 7.0
DB : mariadb 10.10.2
An error occurs when executing prisma db pull in mariadb environment. Why?

Localhost:3306 is operating normally.

tcp        0      0 0.0.0.0:3306            0.0.0.0:*               LISTEN      4822/mariadbd       
```prisma
datasource db {
  provider = "mysql"
  url      = "DATABASE_URL="mysql://user:dbpass@123.123.123.10:3306/dbname"
"
}
```

```bash
npx prisma db pull
Prisma schema loaded from prisma/schema.prisma
Environment variables loaded from .env
Datasource "db": MySQL database "dbname" at "123.123.123.10:3306"

✖ Introspecting based on datasource defined in prisma/schema.prisma

Error: P1001

Can't reach database server at `123.123.123.10`:`3306`

Please make sure your database server is running at `123.123.123.10`:`3306`.
```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: CentOS 7.0
- Database: MariaDB
- Node.js version: v18.17.1

- Run `prisma -v` to see your Prisma version and paste it
prisma                  : 5.4.2
@prisma/client          : 5.4.2
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at ../../../.nvm/versions/node/v18.17.1/lib/node_modules/prisma/node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli ac9d7041ed77bcc8a8dbd2ab6616b39013829574 (at ../../../.nvm/versions/node/v18.17.1/lib/node_modules/prisma/node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.1-2.ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Default Engines Hash    : ac9d7041ed77bcc8a8dbd2ab6616b39013829574
Studio                  : 0.494.0

**Comment by ludralph** [comment, marked as answer]
Hi @sihun0105 👋 

Thank you for raising this question.

Can you make sure that the MariaDB database server at IP address `123.123.123.10` is up and running. You mentioned that localhost:3306 is operating normally, but it's essential to confirm that the remote database is accessible as well. You can use tools like telnet or nc to check if you can establish a network connection to the database server from your CentOS machine.

Example using telnet:
```bash
telnet 123.123.123.10 3306
```

If you cannot establish a connection, it might indicate a network or firewall issue.

**Reply by sihun0105** [reply]
Hi @ludralph 

I found the cause.
prisma does not work in mariadb version 10.10.2.

Although the exact internal factor was not discovered. I found out that the issue occurred due to the version.
I upgraded to 10.10.6 and it was resolved.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
